nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DeduplicateRecord.java

```diff
@@ -625,40 +609,17 @@ public DistributedMapCacheClientWrapper(DistributedMapCacheClient client) {
         @Override
         public boolean contains(String value) {
             try {
-                return client.containsKey(value, STRING_SERIALIZER);
+                return !client.putIfAbsent(value, "", STRING_SERIALIZER, STRING_SERIALIZER);
```

Comment:
This appears to change the behavior, updating the value instead of checking whether the key exists. Can you elaborate on the reasons for this change?

Reply:
I posted some reasons for this change on Slack. The pertinent points are:

> If Cache Identifier is unset, the user must have a separate process to actually put the hashed records to cache, as this processor doesn't currently do that (it only reads). This is arguably counter-intuitive and inconsistent with DetectDuplicate. The reliance on a separate process to write hash entries is problematic as it means:
> Deduplication isn't actually performed on the current FlowFile at all;
> There's a possibility of duplication as we lose atomicity by separating the cache read/write calls; and
> There don't appear to be any record-oriented processors that can conveniently hash each record and put its value to a DMC. To me, this processor should do that, as does DetectDuplicate.

Reply:
Thanks @p-kimberley, I will evaluate the changes soon. For future reference, subsequent commits on the same pull request do not need to be squashed, as it is helpful to evaluate changes from one commit to the next when reviewing.